src/stats.rs

```diff
@@ -287,6 +295,89 @@ impl Reporter {
     }
 }
 
+trait SendStat {
+    fn send_increment(&self, name: &str);
+
+    fn send_count(&self, name: &str, count: i64);
+
+    fn send_time(&self, name: &str, time: u64);
+
+    fn send_gauge(&self, name: &str, value: u64);
+
+    fn submit_count_stat(&self, metric_builder: MetricBuilder<cadence::Counter>) {
+        // TODO: Move tagging logic here
+        if metric_builder.try_send().is_err() {
+            error!("Error sending query metrics to client");
+        }
+    }
+}
+
+impl SendStat for StatsdClient {
+    fn send_increment(&self, name: &str) {
+        let metric_builder: MetricBuilder<cadence::Counter> = self.incr_with_tags(name);
+        self.submit_count_stat(metric_builder);
+    }
+
+    fn send_count(&self, name: &str, count: i64) {
+        let metric_builder = self.count_with_tags(name, count);
+        self.submit_count_stat(metric_builder);
+    }
+
+    fn send_time(&self, name: &str, time: u64) {
+        let metric_builder = self.time_with_tags(name, time);
+        if metric_builder.try_send().is_err() {
+            error!("Error sending query metrics to client");
+        }
+    }
+
+    fn send_gauge(&self, name: &str, value: u64) {
+        let metric_builder = self.gauge_with_tags(name, value);
+        if metric_builder.try_send().is_err() {
+            error!("Error sending query metrics to client");
+        }
+    }
+}
+
+fn new_statsd_client() -> StatsdClient {
+    let config = get_config();
+    if config.general.enable_statsd {
+        let statsd_prefix =
+            env::var("STATSD_PREFIX").expect("Missing STATSD_PREFIX environment variable");
+
+        // Prefer to use socket over udp
+        let sink = match env::var("STATSD_SOCKET") {
+            Ok(statsd_sock) => {
+                let socket = UnixDatagram::unbound().unwrap();
+                socket.set_nonblocking(true).unwrap();
+                let buffered_sink = BufferedUnixMetricSink::from(statsd_sock, socket);
+                QueuingMetricSink::from(buffered_sink)
+            }
+            Err(_) => {
+                let statsd_host =
+                    env::var("STATSD_HOST").expect("Missing STATSD_HOST environment variable");
+                let statsd_port = env::var("STATSD_PORT")
+                    .expect("Missing STATSD_PORT environment variable")
+                    .parse::<u16>()
+                    .unwrap();
+                // Try to create
+                let socket = UdpSocket::bind("0.0.0.0:0").unwrap();
```

Comment:
This panics and crashes the stats collector.

Reply:
So the reason I was okay with having this panic was because the client is only created once and it's done on the [main thread](https://github.com/levkk/pgcat/pull/131/files#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fcR137) before the collector starts so, it'll crash on startup if statsd things aren't configured correctly. If we want to allow the service to continue starting up even if statsd fails then we can make that change to. Depends on the behaviour we want